test/controllers/general/ErrorHandlerSpec.scala

```diff
@@ -41,21 +44,35 @@ class ErrorHandlerSpec extends UnitWithMocksSpec with Stubs with OptionValues wi
     "urls.loginContinue" -> "http://localhost:6791/customs-declare-exports/choice"
   )
 
-  val errorPage = instanceOf[error_template]
-  val appConfig = instanceOf[AppConfig]
-  val request = FakeRequest("GET", "/foo")
+  private val errorPage = instanceOf[error_template]
+  private val appConfig = instanceOf[AppConfig]
+  private val messagesApi = instanceOf[MessagesApi]
 
-  val errorHandler = new ErrorHandler(stubMessagesApi(), errorPage)(appConfig, global)
+  private val request = FakeRequest("GET", "/foo")
+
+  val errorHandler = new ErrorHandler(messagesApi, errorPage)(appConfig, global)
 
   def urlEncode(value: String): String = URLEncoder.encode(value, "UTF-8")
 
   "ErrorHandler.standardErrorTemplate" should {
-    "include title, heading and message" in {
-      val result = errorHandler.standardErrorTemplate("title", "heading", "message")(request).futureValue.body
 
-      result must include("title")
-      result must include("heading")
-      result must include("message")
+    "include default title, heading and message" in {
+      checkView(errorHandler.standardErrorTemplate("global.error.title", "", "")(request).futureValue)
+    }
+
+    "include expected title, heading and message" in {
+      val messages = messagesApi.preferred(request).messages
+      val heading = messages("global.error.heading")
+      val message = messages("global.error.message")
+      checkView(errorHandler.standardErrorTemplate("global.error.title", heading, message)(request).futureValue)
```

Comment:
This test does not seem to prove that you can override the default values as it uses the default values, so is it using the ones specified or the default we can't prove (as they are the same) :thinking:

Reply:
??
The 1st test only use the default messages, passing the `heading` and `message` parameters as empty Strings, then testing the `true` branch at #L62 and #L63, whereas in the 2nd test the 2 parameters' values are retrieved at #L65 and #L66, so testing the `false branch at #L62 and #L63. The fact that we are using the same message keys is not relevant.